playbooks/check_cluster.yml

```diff
@@ -0,0 +1,32 @@
+---
+#- import_playbook: "set_variables.yml"
+
+
+- hosts: ambari-server
+  tasks:
+  - name: Check if a cluster {{ cluster_name }} already exists
+    uri:
+      url: "http://{{ ansible_fqdn }}:8080/api/v1/clusters/{{ cluster_name }}"
+      method: GET
+      force_basic_auth: yes
+      user: "{{ ambari_admin_user }}"
+      password: "{{ ambari_admin_password }}"
+      headers: '{"X-Requested-By":"ambari"}'
+      status_code: -1,200,201,202,404
+    register: cluster_check
+  
+  - name: set skip if exists
+    set_fact:
+      run_ambari_setup: "{{ cluster_check.status == 404 }}"
```

Comment:
Need to change this line to:
``run_ambari_setup: "{{ cluster_check.status == 404 or cluster_check.status == -1 }}"``

Otherwise, it'll skip the HDP install because the previous task returns a status code of -1 if the cluster hasn't been built before.

``
2018-12-11 10:05:00,199 p=9239 u=root |  TASK [Check if a cluster hwx6 already exists] *****************************************************************************************************************************
2018-12-11 10:05:00,199 p=9239 u=root |  task path: /opt/acs/ansible-hortonworks/playbooks/check_cluster.yml:7
2018-12-11 10:05:00,200 p=9239 u=root |  Tuesday 11 December 2018  10:05:00 -0500 (0:00:01.350)       0:02:42.292 ******
2018-12-11 10:05:01,220 p=9239 u=root |  ok: [node1.hwx6.acs.sd.spawar.navy.mil] => {"changed": false, "msg": "Request failed: <urlopen error [Errno 111] Connection refused>", "redirected": false, "status": -1, "url": "http://node1.hwx6.acs.sd.spawar.navy.mil:8080/api/v1/clusters/hwx6"}
2018-12-11 10:05:01,258 p=9239 u=root |  TASK [set skip if exists] *************************************************************************************************************************************************
2018-12-11 10:05:01,258 p=9239 u=root |  task path: /opt/acs/ansible-hortonworks/playbooks/check_cluster.yml:18
2018-12-11 10:05:01,258 p=9239 u=root |  Tuesday 11 December 2018  10:05:01 -0500 (0:00:01.058)       0:02:43.351 ******
2018-12-11 10:05:01,484 p=9239 u=root |  ok: [node1.hwx6.acs.sd.spawar.navy.mil] => {"ansible_facts": {"run_ambari_setup": false}, "changed": false}
``